Assets/Scripts/Timer.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using System;
+
+public class Timer : MonoBehaviour
+{
+    private bool timerActive = false;
+    private bool timerPaused = false;
+
+    private float currentTime;
+    private float pausedTime;
+
+    [SerializeField] private int pauseTime;
+    [SerializeField] private int startSeconds;
+
+    [SerializeField] private Text currentTimeText;
+
+
+    void Start()
+    {
+        currentTime = startSeconds;
+    }
+
+    void Update()
+    {
+        if (StartCheck())
+        {
+            ResetTimer();
+        }
+        else if (IsNotPaused())
+        {
+            currentTime = currentTime - Time.deltaTime;
+        }
+        else if (IsPaused())
+        {
+            pausedTime = pausedTime - Time.deltaTime;
+        }
+        else if (IsPausedTimeElapsed())
+        {
+            timerActive = true;
+            timerPaused = false;
+        }
+
+        TimeSpan time = TimeSpan.FromSeconds(currentTime);
+
+        PrintTime(time);
+
+    }
+    private bool StartCheck()
+    {
+        return timerActive && currentTime <= 0f;
+    }
+    private bool IsNotPaused()
```

Comment:
this sounds weird - and also the rule for booleans is to be named with the true case ---if not is needed - it can be added in the language, for example - !isPaused